tests/OrderableRowsThroughVersionedTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Symbiote\GridFieldExtensions\Tests;
+
+use ReflectionMethod;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldConfig_RelationEditor;
+use Symbiote\GridFieldExtensions\Tests\Stub\ThroughDefiner;
+use Symbiote\GridFieldExtensions\Tests\Stub\ThroughIntermediary;
+use Symbiote\GridFieldExtensions\Tests\Stub\ThroughBelongs;
+use Symbiote\GridFieldExtensions\GridFieldOrderableRows;
+
+class OrderableRowsThroughTest extends SapphireTest
+{
+    protected static $fixture_file = 'OrderableRowsThroughTest.yml';
+
+    protected static $extra_dataobjects = [
+        ThroughDefiner::class,
+        ThroughIntermediary::class,
+        ThroughBelongs::class,
+    ];
+
+    protected static $required_extensions = [
+        ThroughDefiner::class => [ Versioned::class ],
+        ThroughIntermediary::class => [ Versioned::class ],
+        ThroughBelongs::class => [ Versioned::class ],
+    ];
+
+    /**
+     * Mostly the same as GridFieldOrderableRowsTest::testReorderItems
+     * but with some Versioned calls & checks mixed in.
+     */
+    public function testReorderingSaves()
+    {
+        $parent = $this->objFromFixture(ThroughDefiner::class, 'DefinerOne');
+        $sortName = 'Sort';
+
+        $orderable = new GridFieldOrderableRows($sortName);
+        $reflection = new ReflectionMethod($orderable, 'executeReorder');
+        $reflection->setAccessible(true);
+
+        $config = new GridFieldConfig_RelationEditor();
+        $config->addComponent($orderable);
+
+        // This test data is versioned - ensure we're all published
+        $parent->publishRecursive();
+        // there should be no difference between stages at this point
+        foreach ($parent->Belongings() as $item) {
+            if ($item->stagesDiffer()) {
+                $this->fail('Unexpected diference found on stages');
+            }
```

Comment:
Perhaps you could switch this if statement for `$this->assertFalse($item->stagesDiffer(), 'No difference found on stages');` (note typo)